test/builtin/arithmetic/test_basic.py

```diff
@@ -153,8 +153,8 @@ def test_multiply(str_expr, str_expected, msg):
         ("a  b  DirectedInfinity[q]", "a b (q Infinity)", ""),
         # Failing tests
         # Problem with formatting. Parenthezise are missing...
-        #        ("a  b  DirectedInfinity[-I]", "a b (-I Infinity)",  ""),
-        #        ("a  b  DirectedInfinity[-3]", "a b (-Infinity)",  ""),
+        ("a  b  DirectedInfinity[-I]", "a b (-I Infinity)", ""),
```

Comment:
In this branch I am not seeing parenthesis around the result. I assume WMA adds parenthesis? 

I wonder if this is a formatting problem or a precedence problem

Reply:
In WMA
```
In[1]:= a b DirectedInfinity[-I]                                                

Out[1]= a b (-I Infinity)

In[2]:= %//FullForm                                                             

Out[2]//FullForm= Times[a, b, DirectedInfinity[Complex[0, -1]]]

```

On the other hand, in master,

```
In[1]:= a b DirectedInfinity[-I]
Out[1]= a b -I Infinity

In[2]:= %//FullForm
Out[2]//FullForm= Times[a, b, DirectedInfinity[Complex[0, -1]]]

```

Notice that the first output is wrong, and indeed, is a formatting problem.

Reply:
Thanks for the information.  It looks like this is an operator precedence problem.

Reply:
OK, I see the problem now: in this branch, the test is marked as "skip". I solved this precedence issue in another branch, and for some reason, the change was lost. I will try to put this fix and tests in another PR.